src/rust/ide/view/graph-editor/src/component/node/icon/action.rs

```diff
@@ -0,0 +1,186 @@
+use crate::prelude::*;
+
+use ensogl::data::color;
+use ensogl::display::shape::*;
+use ensogl_gui_components::toggle_button::ColorableShape;
+use ensogl_shape_utils::constants::HOVER_COLOR;
+
+
+
+pub mod visibility {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (color_rgba:Vector4<f32>) {
+            let fill_color       = Var::<color::Rgba>::from(color_rgba);
+
+            let width       = Var::<Pixels>::from("input_size.x");
+            let height      = Var::<Pixels>::from("input_size.y");
+            let right_angle = 90.0_f32.to_radians().radians();
+            let unit        = &width/16.0;
+
+            let outer_radius = &unit*5.0;
+
+            let pupil        = Circle(&unit * 1.0);
+            let inner_circle = Circle(&unit * 3.0);
+
+            let outer_circle = Circle(&outer_radius);
+
+            let right_edge   = Triangle(&unit * 7.9, &unit * 4.6);
+            let right_edge   = right_edge.rotate(right_angle);
+            let right_edge   = right_edge.translate_x(&unit * 5.3);
+
+            let left_edge    = right_edge.rotate(2.0 * right_angle);
+
+            let eye_outer    = outer_circle + right_edge + left_edge;
+
+            let eye          = (eye_outer - inner_circle) + pupil;
+            let eye_colored  = eye.fill(fill_color);
+
+            let hover_area   = Rect((width,height)).fill(HOVER_COLOR);
+
+            (eye_colored+hover_area).into()
+        }
+    }
+
+    impl ColorableShape for Shape {
+        fn set_color(&self, color:color::Rgba) {
+            self.color_rgba.set(Vector4::new(color.red,color.green,color.blue,color.alpha));
+        }
+    }
+}
+
+
+pub mod visibility2 {
```

Comment:
These are two versions of "visibility" action? Which one is used? Why we keep the other? It should be explained in a comment.

Additionally public modules should have docs required. Check why it is not required by linter here (or is it?)

Reply:
There were two initial designs proposed, but only one of them is actually used at the moment. I've added a doc to explain.

> Additionally public modules should have docs required. Check why it is not required by linter here (or is it?)

You are right. This mightbe because `node.rs` supresses doc warnings. They need to be manually re-enabled for sub-modules.